src/main/java/io/kokuwa/maven/helm/DependencyBuildMojo.java

```diff
@@ -28,6 +41,180 @@ public class DependencyBuildMojo extends AbstractHelmMojo {
+	 * @param chartFile Chart.yaml file to read
+	 * @return {@link ArrayList}<{@link Dependency}>
+	 * @throws MojoExecutionException
+	 */
+	private ArrayList<Dependency> getDependencies(File chartFile) throws MojoExecutionException {
+		ObjectMapper yamlMapper = new YAMLMapper();
+		yamlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+		Dependencies dependencies;
+		try {
+			dependencies = yamlMapper.readValue(chartFile, Dependencies.class);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read chart dependencies from " + chartFile, e);
+		}
+
+		return dependencies.getDependencies();
+	}
+
+	/**
+	 * Updates the {@link Dependency} with the new overwritten values
+	 * @param dependencies {@Link ArrayList}<{@link Dependency}>
+	 */
+	private void updateDependencies(ArrayList<Dependency> dependencies) {
+		for (Dependency dependency : dependencies) {
+			if (dependency.getRepository() != null && dependency.getRepository().contains("file://")) {
+				dependency.setRepository(overwriteDependencyRepository);
+				dependency.setVersion(overwriteDependencyVersion);
+				dependency.setOverwrite(true);
+			}
+		}
+	}
+
+	/**
+	 * Overwrite the dependencies with their new values to the Chart.yaml 
+	 * @param dependencies {@link ArrayList}<{@link Dependency}>
+	 * @param chartFile Chart.yaml file to read
+	 * @param chartDirectory Directory to write the new Chart.yaml to
+	 * @throws MojoExecutionException
+	 */
+	private void overwriteDependencies(ArrayList<Dependency> dependencies, File chartFile, Path chartDirectory) 
+		throws MojoExecutionException {
+
+		//delete the outdated Chart.lock
+		File chartLock = new File(chartDirectory.toString() + "/Chart.lock");
+		
+		if (chartLock.exists()) {
+			chartLock.delete();
+			getLog().info("Chart.lock deleted at " + chartDirectory);
+		} else {
+			getLog().info("No Chart.lock file found at " + chartDirectory);
+		}
+
+		List<String> overwrittenChart = new ArrayList<>();
+		List<String> originalChart;
+		try {
+			originalChart = Files.readAllLines(Paths.get(chartFile.toURI()), StandardCharsets.UTF_8);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Unable to read Chart.yaml at " + chartDirectory, e);
+		}
+
+		//Used for determining when a dependency has been read
+		boolean name, repository, version;
+		name = version = repository = false;
+
+		int currentDependencyIndex = 0;
+		Dependency currentDependency = dependencies.get(currentDependencyIndex);
+
+		for (String line : originalChart) {
+
+			//all dependencies have been read so just need to write remaining lines
+			if (currentDependencyIndex == dependencies.size()) {
+				overwrittenChart.add(line);
+				continue;
+			}
+
+			//Repository is not a required field so could be null
+			if (currentDependency.getRepository() == null) {
+				repository = true;
+			}
+
+			//check for lines that need to be overwritten
+			if (line.contains(" version:")) { 
```

Comment:
Q: Is the leading space to verify that this is a "nested" version and not the root version for the whole chart? If so, we might look for "dependencies:" to signal that we should be parsing dependencies. Then if/when you encounter another field at the same level as the original "dependencies:", then you're done.  It's worth noting that YAML supports tabs and spaces, and any number of spaces for its indentation blocks

Reply:
Ah actually, strike that. Still a good idea to base the logic around "dependencies:", but YAML does _not_ support tabs for indentation blocks. (At least as of spec 1.2 https://yaml.org/spec/1.2.2/#61-indentation-spaces)

Reply:
Yes the leading space is to verify it's the nested version. I agree looking for dependencies first would be a good addition.